File: lab2/Pair.cpp
#include "Pair.h"

void Pair::SetPair(const int firstInt, const int secondInt) 
{
	_firstInt = firstInt;
	_secondInt = secondInt;
}

int Pair::GetFirstInt() const
{
	return _firstInt;
}

int Pair::GetSecondInt() const
{
	return _secondInt;
}

Pair* Pair::Plus(const Pair& source)
{
	_firstInt += source._firstInt;
	_secondInt += source._secondInt;
	return this;
}

Pair* Pair::Minus(const Pair& source)
{
	_firstInt += source._firstInt;
	_secondInt += source._secondInt;
	return this;
}

File: lab5-6/lab6/lab6/Pair.cs
using System;

namespace lab6
{
    abstract class Pair
    {
        protected int _first = 0;
        protected int _second = 0;

        public int GetFirst
        {
            get
            {
                return _first;
            }
        }
        public int GetSecond
        {
            get
            {
                return _second;
            }
        }

        public Pair(int first, int second)
        {
            _first = first;
            _second = second;
        }

        abstract public Pair Plus(Pair source);
        abstract public Pair Minus(Pair source);
        abstract public string GetPair();

    }
}
